src/async_hta_service.hpp

```diff
@@ -442,6 +450,89 @@ class AsyncHtaService
         return it.first->second;
     }
 
+private:
+    class Stats
+    {
+    public:
+        template <typename T>
+        void add_read_duration(T duration)
+        {
+            {
+                std::scoped_lock lock(stats_lock_);
+                read_count_++;
+                read_duration_ +=
+                    std::chrono::duration_cast<std::chrono::duration<double>>(duration).count();
+            }
+            log_stats_();
+        }
+
+        template <typename T>
+        void add_write_duration(T duration)
+        {
+            {
+                std::scoped_lock lock(stats_lock_);
```

Comment:
You don't need a `scoped_lock`, which is required for more than one mutexs. `lock_guard` is die idiomatic tool